JFrog cloud (#2212)
DataDog/integrations-extras: jfrog_platform_cloud/README.md

# Agent Check: JFrog Platform Cloud

## Overview

> [JFrog](https://jfrog.com/) is a universal hybrid, end-to-end DevOps platform. JFrog Artifactory is the single solution for housing and managing all the artifacts, binaries, packages, files, containers, and components for use throughout your software supply chain.

Reviewer: ```suggestion
[JFrog][1] is a universal hybrid, end-to-end DevOps platform. JFrog Artifactory is the single solution for housing and managing all the artifacts, binaries, packages, files, containers, and components for use throughout your software supply chain.
```

JFrog Artifactory serves as your central hub for DevOps, integrating with your tools and processes to improve automation, increase integrity, and incorporate best practices along the way.

> JFrog's SaaS Log Streamer is a log streaming solution built by JFrog for SaaS customers. This solution will stream JFrog Artifactory logs from the customer's JFrog SaaS instance straight into their Datadog instance.
> 
> Customers who use both JFrog and Datadog will be able to visualize Artifactory logs inside pre-configured Datadog dashboards. This integration also has built-in support for Datadog log pipelines which means logs streamed from JFrog will be preprocessed and automatically converted into the Datadog log format, allowing teams to uniquely name logs per their needs, drill down into Artifactory logs through searchable facets, and easily monitor their JFrog SaaS instance.
> 
> At the initial release this integration will stream the following artifactory logs to Datadog:

Reviewer: ```suggestion
JFrog's SaaS Log Streamer is a log streaming solution built by JFrog for SaaS customers. This solution  streams JFrog Artifactory logs from the customer's JFrog SaaS instance into their Datadog instance.

Customers who use both JFrog and Datadog can visualize Artifactory logs with out-of-the-box Datadog dashboards. This integration also supports Datadog log pipelines, which means logs streamed from JFrog are preprocessed and automatically converted into the Datadog log format, allowing teams to uniquely name logs per their needs, drill down into Artifactory logs through searchable facets, and easily monitor their JFrog SaaS instance.

This integration enables you to stream the following artifactory logs to Datadog:
```

- **access-audit.log**
- **artifactory-request.log**
- **access-security-audit.log**

> These logs will allow customers to readily know who accessed what repositories and how often. The logs will also show what IP addresses accessed those repositories. Log types such as traffic.log, artifactory-access.log and more request logs will be added to this integration in future updates.
> 
> JFrog’s SaaS Log Streaming is currently in Beta. While in Beta, the cloud log streaming feature will only be available inside the MyJFrog portal for select JFrog Enterprise + customers. JFrog plans to GA this feature later in Q2 2024 at which point it will be available to all JFrog Enterprise + customers.

Reviewer: We don't typically talk about what features/product limitations will be delivered in future updates. Is it fine to leave this out and add this info back once this becomes available later on?

```suggestion
These logs allow you to readily know who accessed what repositories and how often. The logs also show what IP addresses accessed those repositories. 

JFrog’s SaaS Log Streaming is in beta. While in beta, the cloud log streaming feature will only be available inside the MyJFrog portal for select JFrog Enterprise + customers. 
```

## Setup

> **Note:** The integration requires JFrog Enterprise Plus subscription. 

Reviewer: ```suggestion
This integration requires a JFrog Enterprise Plus subscription. 
```

### Installation

> Create [DataDog API key](https://app.datadoghq.com/organization-settings/api-keys).  

Reviewer: ```suggestion
Create a [Datadog API key][2].  
```

> On the [MyJFrog Portal](https://my.jfrog.com), go to Settings -> JFrog Cloud Log Streaming - BETA, and enable the Log Streamer.
> Select Datadog as the vendor. Add your Datadog API key, the Datadog intake URL for your [Datadog site](https://docs.datadoghq.com/getting_started/site/) (`https://http-intake.logs.datadoghq.com/api/v2/logs`), and `ddtags`. Click Save.
> Your logs will start streaming into Datadog in 24 hours or less.

Reviewer: ```suggestion
On the [MyJFrog Portal][4], go to Settings -> JFrog Cloud Log Streaming - BETA, and enable the Log Streamer.

Select Datadog as the vendor. Add your Datadog API key, the Datadog intake URL for your [Datadog site][3] (for example: `https://http-intake.logs.datadoghq.com/api/v2/logs`), and `ddtags`. Click **Save**.

Your logs will start streaming into Datadog in 24 hours or less.
```

## Support

> Need help? Contact [JFrog Support](https://support.jfrog.com/).

Reviewer: ```suggestion
Need help? Contact [JFrog Support][5].

[1]: https://jfrog.com/
[2]: https://app.datadoghq.com/organization-settings/api-keys
[3]: https://docs.datadoghq.com/getting_started/site/
[4]: https://my.jfrog.com
[5]: https://support.jfrog.com/
```
